### PR TITLE
Plugins: Add plugin resource tests

### PR DESCRIPTION
**Pull request by wbrowne**
Note: I removed 

```go
config := httpadapter.PluginConfigFromContext(req.Context())
```

from the test response payload as an error was being thrown when trying to marshal a plugin context that has `json.RawMessage{}` (IE empty) for app instance settings. I don't think plugin context is ever intended to be marshalled to JSON so it's good practice to remove this regardless

### CODE REVIEW

File: pkg/api/plugin_resource_test.go [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+	"github.com/grafana/grafana/pkg/services/user"
+	"github.com/grafana/grafana/pkg/setting"
+	"github.com/grafana/grafana/pkg/tsdb/cloudwatch"
+	"github.com/grafana/grafana/pkg/tsdb/testdatasource"
+	"github.com/grafana/grafana/pkg/web/webtest"
+)
+
+func TestCallResource(t *testing.T) {
+	if testing.Short() {
+		t.Skip("skipping integration test")
```
Comment (marefr) [inline pkg/api/plugin_resource_test.go:46]: Is this  an integration test? Should test be named `TestIntegrationCallResource`then?

**Review by marefr — APPROVED** [review]
LGTM with a comment

File: pkg/api/plugin_resource_test.go (hunk 1 above)
Comment (wbrowne) [reply pkg/api/plugin_resource_test.go]: Good catch! Nah it's more testing API layer so will remove this

**Review by marefr — APPROVED** [review]
LGTM

**Pull request merged**